### PR TITLE
Migrate `restoreExperiment` endpoint.

### PR DESCRIPTION
**Pull request by dsuhinin**
PR related to -> https://github.com/users/jgiannuzzi/projects/2?pane=issue&itemId=72542972

### CODE REVIEW

File: pkg/contract/service/tracking.g.go
Comment (jgiannuzzi) [inline pkg/contract/service/tracking.g.go:1]: please don't reformat the `*.g.go` files, as they are automatically generated

File: pkg/tracking/store/mock_tracking_store.go
Comment (jgiannuzzi) [inline pkg/tracking/store/mock_tracking_store.go:1]: please use the version of mockery that comes with the devcontainer

**Review by jgiannuzzi — COMMENTED** [review]
please add the Python store bindings too

**Comment by nojaf** [comment]
> please add the Python store bindings too

@jgiannuzzi probably means adding the correct entry in https://github.com/jgiannuzzi/mlflow-go/blob/main/mlflow_go/store/tracking.py?

File: pkg/tracking/store/sql/experiments.go [hunk 1]
```diff
@@ -134,6 +134,62 @@ func (s TrackingSQLStore) DeleteExperiment(id string) *contract.Error {
...
+		return contract.NewErrorWith(
+			protos.ErrorCode_INVALID_PARAMETER_VALUE,
+			fmt.Sprintf("failed to convert experiment id (%s) to int", id),
+			err,
+		)
+	}
+
+	if err := s.db.Transaction(func(transaction *gorm.DB) error {
+		// Update experiment
+		uex := transaction.Model(&models.Experiment{}).
```
Comment (nojaf) [inline pkg/tracking/store/sql/experiments.go:149]: I believe the Python implementation will only update when the experiment has `ViewType.DELETED_ONLY`. In the Go implementation you could restore an experiment that was not deleted? Or is this checked elsewhere?

**Review by nojaf — COMMENTED** [review]
Great work! Solid PR, I only have one small remark about a potential slight deviation from the Python code.

@jgiannuzzi we lack some CI here 😔, would be great if we could at least run the tests in some runner.

**Review by nojaf — DISMISSED** [review]
Looks good to me!

**Review by nojaf — APPROVED** [review]

**Pull request merged**